lib/blockcache/blockcache.go

```diff
@@ -280,25 +280,42 @@ func (c *cache) cleanByTimeout() {
 func (c *cache) GetBlock(k Key) Block {
 	c.requests.Add(1)
 	var e *cacheEntry
-	c.mu.Lock()
-	defer c.mu.Unlock()
+
+	c.mu.RLock()
```

Comment:
This change smells like an introduction of a data race: what happens when the first goroutine updates `c.m` under `write` lock, while other goroutines read `c.m` under `read` lock?